src/AppBundle/Admin/AdministratorAdmin.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace AppBundle\Admin;
+
+use AppBundle\Entity\Administrator;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+
+class AdministratorAdmin extends AbstractAdmin
+{
+    /**
+     * @var EncoderFactoryInterface
+     */
+    private $encoders;
+
+    /**
+     * @param Administrator $admin
+     */
+    public function prePersist($admin)
+    {
+        parent::prePersist($admin);
+
+        $admin->setPassword($this->encoders->getEncoder($admin)->encodePassword($admin->plainPassword, null));
```

Comment:
I don't think it's a good idea to have a public property for the plain password, it may change in this listener at any time.

The password should be encoded in the factory or a manager only. And this listener should only force `null` on this public property, not encode it.

What do you think? /cc @hhamon